### ISSUE TITLE
  MozFest 2015 site: View proposals UI

### ISSUE THREAD

**Issue by edrushka** [mozilla/2015.mozillafestival.org#22]
Wireframes: coming soon

#### Functionality
MVP:
- staff can edit data (this doesn't need beautiful UI)
- people can view proposed submissions

Nice-to Have:
- people can shares links to proposed submissions

**Comment by edrushka** [comment]
@adiladiladil we are going to move ahead with this as well. @iamjessklein is going to finalize the wireframe and @saallen will provide a list of filters to search by.

**Comment by Saallen** [comment]
Do not have full list of filters but could we start with 
* Lead
* Teach
* Youth

**Comment by iamjessklein** [comment]
I started doing very complicated wireframes for the view proposals, but the I started to wonder - what if the proposals all went into github? Cascadia [does a really nice job](https://github.com/cascadiajs/2015.cascadiajs.com/issues?q=is%3Aissue+is%3Aclosed) of having github handle the proposal process, and so does Brooklyn.js. This would allow us to easily add or delete tags, link out to the proposals, give authors the ability to edit and reassign proposals from proposals to sessions. 

Thoughts @Saallen @edrushka @cassiemc

**Comment by Saallen** [comment]
I have two questions around this
* How does this work on the website? Do they submit directly onto Github?
* How do we upload the selected submission to the schedule doc or to an excel sheet. Is this a manual process?

**Comment by iamjessklein** [comment]
My thinking is that the flow could work like this:
* participants fill out the form on the festival site
* the contents of the form autocomplete into a github issue with tags
* mozfest staff can review and sort forms via tags
* when sessions are chosen, they can be added to a track (or theme?) tag or milestone
* During the week of the event, the schedule section of the site will display the schedule and then link out to the individual sessions issues
* Session hosts have the option to use their github issue to update the status on their session or use it for documentation - and participants can add to it.
* After the event, issues will get archived - or - if they are active.. they will act as a project management tool.

**Comment by edrushka** [comment]
super helpful, Jess. Flagging @ScottDowne

**Comment by ScottDowne** [comment]
I actually like pushing and storing these in github issues, what @iamjessklein is saying makes a lot of sense. We should push to do it this way.

At the very least it's as easy to setup as google forms, and as functional.

It has the benefit of better sorting and general usability.

We get editing (via github issues) for free.

It also gives us the option to connect to it later programmatically and do more with it, if we need to, where as google forms is kinda locked into a limited feature set.

Not sure if the issues can be viewed as a spreadsheet, but instead an issue list.

**Comment by edrushka** [comment]
Is there an existing widget to export issues to a spreadsheet? I believe the team requires an Excel/google spreadsheet. What are you thinking, @Saallen?

**Comment by Saallen** [comment]
Yes, in order for us to use any schedule app we need to be able to export all sessions ( ones which we would tag as festival sessions , as we wont need to move all) onto an excel sheet with certain fixed headings. 
On the website those who have submitted sessions like to see them on the website and it provides a social media buzz if they can tweet or link about the sessions.
On the back end all Space Wranglers would need to see their relevantly tagged sessions as they read through and the ones which will take place at the festival.
I think because of my very hacked knowledge of Github I have alot of questions around this filing type of 600+ sessions and would love to be walked through it on one the github links jess ref above.

**Comment by ScottDowne** [comment]
@Saallen Not sure if you got answers to these two questions:

    How does this work on the website? Do they submit directly onto Github?
    How do we upload the selected submission to the schedule doc or to an excel sheet. Is this a manual process?

Asking people to go to github to propose a session isn't really an option, I see this all happen from inside the mozfest site.

It wouldn't super difficult to add something to the site to exports as a spreadsheet.

I'm not super clear on why any schedule app needs it to be in a spreadsheet. Is the schedule app something we're creating or are we using an existing one that needs to be a spreadsheet?

If we're creating it, I would rather the data be exported as json, much easier for the front end to consume that, and it's how browsers natively deal with data. Converting json to a spreadsheet can be done.

**Comment by edrushka** [comment]
@ScottDowne - Sarah and I just had a big talk about this and we're cautiously excited about this idea :relieved: Thanks, @iamjessklein for the suggestion!

Can we move this convo to the [scheduling tool ticket] (https://github.com/MozillaFoundation/plan/issues/442)? I think the next step is to explore the shortlist for the scheduling tools, and investigate if we can import/export between github and the tool.

In terms of UI, I think @adiladiladil should continue with the frontend design he's working on [here] (https://redpen.io/xv7a70e3af17bd08fa).

**Comment by edrushka** [comment]
No UI for viewing the submitted proposals yet. This is waiting on #27
